clients/src/test/java/org/apache/kafka/common/record/MemoryRecordsTest.java

```diff
@@ -366,6 +406,10 @@ public void testFilterToPreservesProducerInfo() {
             assertEquals(baseSequence1, secondBatch.baseSequence());
             assertEquals(baseSequence1 + 2, secondBatch.lastSequence());
             assertFalse(secondBatch.isTransactional());
+            List<Record> secondBatchRecords = TestUtils.toList(secondBatch);
+            assertEquals(2, secondBatchRecords.size());
+            assertEquals(baseSequence1 + 1, secondBatchRecords.get(0).sequence());
+            assertEquals(baseSequence1 + 2, secondBatchRecords.get(1).sequence());
```

Comment:
Nit: probably worth checking the key and offset of the record as well, just to close the loop. Similarly for the other places where we are checking the record values.